app/tests/pages/index.test.tsx

```diff
@@ -1,25 +1,35 @@
-// test/pages/index.test.js
-import { render, screen, waitFor } from '@testing-library/react'
-import { axe } from 'jest-axe'
-
 import Index from '@pages/index'
 
-describe('Index', () => {
-  it('should render the heading', () => {
-    render(<Index />)
+import { setup } from '../helpers/setup'
+import {
+  testAccessibility,
+  testActionButtonRoute,
+  testSnapshot,
+} from '../helpers/sharedTests'
+
+/**
+ * Test setup
+ */
 
-    const heading = screen.getByText(/Start an application for WIC/i)
+const route = '/'
+const forwardRoute = '/how-it-works'
 
-    expect(heading).toBeInTheDocument()
-    expect(heading).toMatchSnapshot()
-  })
+/**
+ * Begin tests
+ */
 
-  it('should pass accessibility scan', async () => {
-    const { container } = render(<Index />)
-    const results = await axe(container)
+it('should match full page snapshot', () => {
+  setup(route)
+  testSnapshot(<Index forwardRoute={forwardRoute} />)
```

Comment:
this testSnapshot abstraction is unique to me - my initial reaction is I'm not sure how much value it adds in terms of making code more dry etc vs the cost of knowing on read what is being tested. 

I know the snapshot testing against the tree is an example in the jest docs but fwiw I am wondering the merits of that vs testing against container returned from the render, which I've seen more frequently recently.

Reply:
@aligg Very useful feedback. Thank you!

I had a clarifying question about this:

> I know the snapshot testing against the tree is an example in the jest docs but fwiw I am wondering the merits of that vs testing against container returned from the render, which I've seen more frequently recently.

Do you mean the merits of testing like I am doing here:

```
  const tree = renderer.create(element).toJSON()
  expect(tree).toMatchSnapshot()
```

vs doing a test that doesn't use snapshot testing?

Or do you mean that there is value in doing snapshot testing and there a different way of creating jest snapshots that doesn't use `renderer.create(element).toJSON()` and uses `render` instead?

Reply:
the latter!

I am most familiar with e.g.

```
import { render } from "@testing-library/react";

describe("blahblahtest", () => {
  it("should render as blahblah", () => {
    const { container } = render(<Component />);
    expect(container).toMatchSnapshot();
  });
});
```

You can read the source code for testing-library render [here](https://github.com/testing-library/react-testing-library/blob/fe163763bba28bbd8ec08f94e36db40efbdd2ae8/src/pure.js) - i haven't looked at it in detail recently but IIRC it's already returning from render some json you can test against - so I think you could just rely on render if you want! 

I know for example on PFML in the portal we did not use react-test-renderer at all that I know of.

Reply:
Thank you!